lib/ts/ui/uiutils.tsx

```diff
@@ -0,0 +1,22 @@
+import type { RecipeFeatureComponentMap } from "../types";
+
+// eslint-disable-next-line comma-spacing
+export const getRecipeFeaturesSSRSafe = <RecipeInstance,>(
+    getRecipeInstance: () => RecipeInstance,
+    getRecipefeatures: (recipeInstance: RecipeInstance) => RecipeFeatureComponentMap
+) => {
+    let recipeInstance: RecipeInstance;
+
+    try {
+        recipeInstance = getRecipeInstance();
+    } catch (e) {
+        // eslint-disable-next-line supertokens-auth-react/no-direct-window-object
+        if (typeof window === "undefined") {
+            return {};
+        }
+
+        throw e;
+    }
```

Comment:
While this solves the immediate issue, I think I'm not sure how much this renders in SSR (if at all).
- This change means that we are adding different routes (and components) during SSR and browser rendering. This could cause hydration issues
- Please check that at least some SSR happens (e.g.: by checking the preview of the html loaded by during the initial page load of `/auth`)